### ISSUE TITLE
Unable to change an existing bit fields default value in mssql

### ISSUE THREAD

**Issue by bergheim** [typeorm/typeorm#5516]
**Issue type:**
[x] bug report

**Database system/driver:**

[x] `mssql`

**TypeORM version:**

[x] `latest`

I am trying to change a default value for a field, but end up getting "Column already has a DEFAULT bound to it", because the old one does not seem to be removed.

Below is an example that should trigger the error.

**Steps to reproduce or a small repository showing the problem:**

```ts
async function main() {
    const entities = [new EntitySchema({
        name: 'foo',
        columns: {
            'id': {
                type: 'uuid',
                primary: true,
                generated: 'uuid',
            },
            'boolfield': {
                type: 'bit',
                default: 1,
            },
        },
    })];

    const options: ConnectionOptions = {
        type: 'mssql',
        host: 'some.host',
        username: 'some.user',
        password: 'some.password',
        database: 'some.database',
        logging: true,
        synchronize: true,
        entities,
        extra: {
            options: {
                encrypt: true,
            }
        },
    };

    const connection = await createConnection(options);
    const qr = await connection.createQueryRunner();
    const table = await qr.getTable('foo');
    const existingCol = table.columns.find(c => c.name === 'boolfield');

    await qr.changeColumn(
        table,
        existingCol,
        new TableColumn({
            name: 'boolfield',
            type: 'bit',
            default: 0, // this will trigger the error
        }),
    );

    // never reached
    console.log('great success');

    // @ts-ignore
    process.exit(0);
}

main();

```

**Comment by bergheim** [comment]
Fixed in 9fc8329f323094. However, if you rename the field at the same time, the drop contraint statement fails:

```ts
async function main() {
    const entities = [new EntitySchema({
        name: 'foo',
        columns: {
            'id': {
                type: 'uuid',
                primary: true,
                generated: 'uuid',
            },
            'boolfield': {
                type: 'bit',
                default: 1,
            },
        },
    })];

    const options: ConnectionOptions = {
        type: 'mssql',
        host: 'asdf',
        username: 'asdf',
        password: 'asdf',
        database: 'asdf',
        logging: true,
        synchronize: true,
        entities,
        extra: {
            options: {
                encrypt: true,
            }
        },
    };

    const connection = await createConnection(options);
    const qr = await connection.createQueryRunner();
    const table = await qr.getTable('foo');
    const existingCol = table.columns.find(c => c.name === 'boolfield');

    await qr.changeColumn(
        table,
        existingCol,
        new TableColumn({
            name: 'boolfieldrenamed', // now this will fail
            type: 'bit',
            default: 0,
        }),
    );

    console.log('great success');

    // @ts-ignore
    process.exit(0);
}

main();
```

**Comment by bergheim** [comment]
Confirmed fixed in 3a9c4e0
